homeassistant/components/fibaro/__init__.py

```diff
@@ -357,39 +355,74 @@ def _read_devices(self):
                 pass
 
 
-def setup(hass: HomeAssistant, base_config: ConfigType) -> bool:
-    """Set up the Fibaro Component."""
+async def async_setup(hass: HomeAssistant, base_config: ConfigType) -> bool:
+    """Migrate configuration from configuration.yaml."""
+    if DOMAIN not in base_config:
+        return True
     gateways = base_config[DOMAIN][CONF_GATEWAYS]
-    hass.data[FIBARO_CONTROLLERS] = {}
+    if gateways is None:
+        return True
+
+    # check if already configured
+    if hass.config_entries.async_entries(DOMAIN):
+        return True
 
-    def stop_fibaro(event):
-        """Stop Fibaro Thread."""
-        _LOGGER.info("Shutting down Fibaro connection")
-        for controller in hass.data[FIBARO_CONTROLLERS].values():
-            controller.disable_state_handler()
+    for gateway in gateways:
+        # prepare new config based on configuration.yaml
+        conf = {}
+        conf[CONF_URL] = gateway[CONF_URL]
+        conf[CONF_USERNAME] = gateway[CONF_USERNAME]
+        conf[CONF_PASSWORD] = gateway[CONF_PASSWORD]
+        conf[CONF_IMPORT_PLUGINS] = gateway[CONF_PLUGINS]
```

Comment:
```suggestion
        conf = {
            CONF_URL: gateway[CONF_URL],
            CONF_USERNAME: gateway[CONF_USERNAME],
            CONF_PASSWORD: gateway[CONF_PASSWORD],
            CONF_IMPORT_PLUGINS: gateway[CONF_PLUGINS]
        }
```

Its a bit more efficient to do it in one shot